src/routes/blogs.rs

```diff
@@ -96,8 +98,27 @@ pub fn new(rockets: PlumeRocket, _user: User) -> Ructe {
 }
 
 #[get("/validate/<validation_id>")]
-pub fn validate(validation_id: String, rockets: PlumeRocket) -> Result {
-    unimplemented!("No idea what to do here yet")
+pub fn domain_validation(
+    validation_id: String,
+    valid_domains: State<HashMap<&str, Instant>>,
+) -> Result<String, String> {
+    let value = valid_domains.inner().get(validation_id.as_str());
+    if value.is_none() {
+        return Err(String::from("404"));
+    }
+
+    // we have valid id, now check the time
+    let valid_until = value.unwrap();
+    let now = Instant::now();
+
+    // nope, expired (410: gone)
+    if now.checked_duration_since(*valid_until).unwrap().as_secs() > 0 {
```

Comment:
you should use `duration_since` instead, as you're unwrapping it it's doing exactly the same thing, and it's stable